pages/api/delegates/delegates.ts

```diff
@@ -0,0 +1,57 @@
+/*
+
+SPDX-FileCopyrightText: © 2023 Dai Foundation <www.daifoundation.org>
+
+SPDX-License-Identifier: AGPL-3.0-or-later
+
+*/
+
+import { NextApiRequest, NextApiResponse } from 'next';
+import { fetchDelegatesPaginated } from 'modules/delegates/api/fetchDelegates';
+import { DelegatesAPIResponse } from 'modules/delegates/types';
+import { DEFAULT_NETWORK, SupportedNetworks } from 'modules/web3/constants/networks';
+import withApiHandler from 'modules/app/api/withApiHandler';
+import validateQueryParam from 'modules/app/api/validateQueryParam';
+
+export default withApiHandler(async (req: NextApiRequest, res: NextApiResponse<DelegatesAPIResponse>) => {
+  const network = validateQueryParam(req.query.network, 'string', {
+    defaultValue: DEFAULT_NETWORK.network,
+    validValues: [SupportedNetworks.GOERLI, SupportedNetworks.GOERLIFORK, SupportedNetworks.MAINNET]
+  }) as SupportedNetworks;
+
+  const first = validateQueryParam(req.query.first, 'number', {
+    defaultValue: 20,
+    minValue: 1,
+    maxValue: 30
+  }) as number;
+
+  const after = validateQueryParam(req.query.after, 'string', {
+    defaultValue: null
+  }) as string | null;
+
+  const includeExpired = validateQueryParam(req.query.includeExpired, 'boolean', {
+    defaultValue: false
+  }) as boolean;
+
+  const orderBy = validateQueryParam(req.query.orderBy, 'string', {
+    defaultValue: 'DATE',
+    validValues: ['MKR', 'DELEGATORS', 'DATE']
+  }) as string;
+
+  const orderDirection = validateQueryParam(req.query.orderDirection, 'string', {
+    defaultValue: 'DESC',
+    validValues: ['ASC', 'DESC']
+  }) as string;
+
+  const status = validateQueryParam(req.query.status, 'string', {
+    defaultValue: 'ALL',
+    validValues: ['RECOGNIZED', 'SHADOW', 'ALL']
+  }) as string;
+
+  const validatedQueryParams = { network, first, after, includeExpired, orderBy, orderDirection, status };
```

Comment:
use the same standard of pagination that is found across the api (page / pageSize)

the start/ first are more related to the graphql implementation. The API should transform page/pageSize to whatever is needed by the graphql backend

Reply:
Are you referring to only the names of the parameters or that we should change `after: Cursor` for `page: number`? The GraphQL API generates a random Cursor that is used for pagination, which would be a bit difficult to change to a page number, unless we stored the cursor somewhere after each GraphQL response

Reply:
oh, interesting. can you explain me what are the benefits of using the cursor versions increasing the start/limit values?

Reply:
Actually, I read about a lot of benefits of cursor pagination over offset pagination, but doing a deeper investigation seems like most of them don't really apply to us :/
- I don't think we can avoid the situation where the API endpoint would return some duplicate or missing entries as we're ordering the rows at the time of execution (based on MKR or delegator count), so if the delegations change, a delegate could be missed on one of the pages. That said, I think that's unlikely to happen to recognized delegates as all of them will be returned on the first or second page.
- Cursor pagination is claimed to be faster as the SQL query doesn't have to iterate over the full table, but just select the requested number of records after the provided unique identifier, but despite Postgraphile returning cursors on paginated queries, it is actually just running offset pagination in the background, so the performance improvements cannot be applied here.

I'll switch the API to use offset for pagination (which would allow us to use page numbers), it would involve adding a couple of extra lines of code but I think it'll be easier to interact with.

Reply:
cool, agree with the points, and the pagination with changing data is something we can not avoid, is what it is